### PR TITLE
feat(broccoli): add jade render

### PR DESCRIPTION
**Pull request by t-fox**

### CODE REVIEW

**Comment by t-fox** [comment]
I signed it!

**Comment by NoNameProvided** [comment]
Jade was renamed to pug, the transition is in progress, but I think the word `jade` should not be used anymore and  it should be replaced with `pug` everywhere.

**Comment by hansl** [comment]
Hi.  Thanks @t-fox!  Could you add tests? You can base those on the CSS ones.  Also, seems like linting was failing. Please fix and rebase :)

File: lib/broccoli/angular-broccoli-jade.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  jade = requireOrNull(`${process.env.PROJECT_ROOT}/node_modules/jade`);
+}
+
+class JadePlugin extends Plugin {
+  constructor(inputNodes, options) {
+    super(inputNodes, {});
+
+    options = options || {};
+    Plugin.call(this, inputNodes, {
+      cacheInclude: [/\.jade/]
```
Comment (gkalpak) [inline lib/broccoli/angular-broccoli-jade.js]: Shouldn't this be `/\.jade$/` (with a `$` at the end) ?

File: lib/broccoli/angular-broccoli-jade.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    });
+    this.options = options;
+  }
+
+  build() {
+    this.listFiles().forEach(fileName => {
+      this.compile(path.normalize(fileName),
+        path.normalize(this.inputPaths[0]),
+        path.normalize(this.outputPath));
+    })
```
Comment (gkalpak) [inline lib/broccoli/angular-broccoli-jade.js]: Missing semicolon.

**Comment by t-fox** [comment]
OK, @hansl. Added test and rebased. ;) It also uses pug instead of jade.

**Comment by filipesilva** [comment]
@t-fox can you add some instructions to the readme regarding how to use this functionality?

File: README.md [hunk 3]
```diff
@@ -220,6 +220,9 @@ To use one just install for example `npm install node-sass` and rename `.css` fi
 
 The `Angular2App`'s options argument has `sassCompiler`, `lessCompiler`, `stylusCompiler` and `compassCompiler` options that are passed directly to their respective CSS preprocessors.
 
+### Additional Support
+- pug (pug)
```
Comment (filipesilva) [inline README.md]: What does this section mean?

**Comment by t-fox** [comment]
@filipesilva I updated readme and rebased

**Comment by krasevych** [comment]
What about generate code with `.pug` ?
Because generate code with *.html and then rename to *.pug is not good(

**Comment by t-fox** [comment]
@krasevych Isn't it obvious? Nonetheless I updated Readme.

**Comment by ivanmayes** [comment]
+1 for this

File: tests/e2e/e2e_workflow.spec.js [hunk 4]
```diff
@@ -367,6 +367,39 @@ describe('Basic end-to-end Workflow', function () {
     });
   });
 
+  it.skip('Installs pug support successfully', function() {
```
Comment (caseyhoward) [inline tests/e2e/e2e_workflow.spec.js]: Why is this skipped?

Comment (t-fox) [inline tests/e2e/e2e_workflow.spec.js]: Other tests like sass support are skipped too since 7c0ba39f0cfc4af7a8e1f6f6d364d741870dc200

**Comment by ShadowManu** [comment]
Besides updating, what's missing for the release of this?

**Comment by ganySA** [comment]
Has this been included?

**Comment by ShadowManu** [comment]
It's still marked as a work in progress. I feel I'm still too green to contribute, but I could help out with what's missing, if original contributors can't.

**Comment by ShadowManu** [comment]
@filipesilva can this issue be revived?

**Comment by filipesilva** [comment]
Closed as this PR was made obsolete by https://github.com/angular/angular-cli/pull/1455. 

I appreciate all the hard work that @t-fox put into this, but we couldn't port it into the new webpack build system as is so we decided not to merge once we started working on webpack.

**Comment by niklas-dahl** [comment]
@filipesilva since the webpack branch is merged now, is there a plan to add a pug/jade loader, just like there is already a sass/less/stylus loader?

It would be awesome if we could just write:
```
@Component({
  selector: 'app-root',
  templateUrl: './app.component.pug',
  styleUrls: ['./app.component.scss']
})
```

**Comment by tycho01** [comment]
@niklas-dahl: this makes some sense; the Angular team [does](https://github.com/angular/angular.io/tree/master/public) in fact like it themselves. :)

**Pull request closed without merging**